src/libifconfig_internal.c

```diff
@@ -39,6 +40,19 @@
 #include "libifconfig_internal.h"
 
 int
+ifconfig_getifaddrs(ifconfig_handle_t *h)
+{
+	int ret;
+
+	if (h->ifap == NULL) {
+		ret = getifaddrs(&h->ifap);
+		/*h->ifa_cursor = h->ifap;*/
+		return (ret);
+	} else
```

Comment:
I agree. It's not likely to be a problem for applications that run for less than a few seconds (ifconfig?), but libifconfig may be used in applications that have a persistent state.